batch/indexer_Position_Bond.py

```diff
@@ -430,47 +462,257 @@ def __sync_approve_transfer(self, block_from: int, block_to: int):
+                    )   
+                    from_exchange_commitment = Contract.call_function(
+                        contract=exchange,
+                        function_name="commitmentOf",
+                        args=(from_account_address, token_address,),
                         default_returns=0
+                    )  
+                    self.sink.on_position(
+                        token_address=token_address,
+                        account_address=from_account_address,
+                        exchange_balance=from_exchange_balance,
+                        exchange_commitment=from_exchange_commitment
                     )
-                    to_pending_transfer = Contract.call_function(
-                        contract=token,
-                        function_name="pendingTransfer",
-                        args=(to_account,),
+                    to_exchange_balance = Contract.call_function(
+                        contract=exchange,
+                        function_name="balanceOf",
+                        args=(to_account_address, token_address,),
+                        default_returns=0
+                    )   
+                    to_exchange_commitment = Contract.call_function(
+                        contract=exchange,
+                        function_name="commitmentOf",
+                        args=(to_account_address, token_address,),
+                        default_returns=0
+                    )  
+                    self.sink.on_position(
+                        token_address=token_address,
+                        account_address=to_account_address,
+                        exchange_balance=to_exchange_balance,
+                        exchange_commitment=to_exchange_commitment
+                    )
+
+            except Exception as e:
+                LOG.exception(e)
+
+    def __sync_set_commitment(self, block_from: int, block_to: int):
+        """[DEX]Sync Events Involved setCommitment
+
+        :param block_from: From block
+        :param block_to: To block
+        :return: None
+        """
+        for exchange in self.exchange_list:
+            try:
+                order_events = []
+                order_events.append(exchange.events.NewOrder.getLogs(
+                    fromBlock=block_from,
+                    toBlock=block_to
+                ))
+                order_events.append(exchange.events.CancelOrder.getLogs(
+                    fromBlock=block_from,
+                    toBlock=block_to
+                ))
+                order_events.append(exchange.events.ForceCancelOrder.getLogs(
+                    fromBlock=block_from,
+                    toBlock=block_to
+                ))
+                target_events = []
+                for event in order_events:
+                    if token_address in self.token_address_list:
+                        target_events.append(event)
+                for event in target_events:
```

Comment:
Need to sort the merged list by blockNumber and logIndex.

Reply:
As you wish.
fixed !